src/conversion.jl

```diff
@@ -40,33 +41,72 @@ GI.getfeature(::Any, d::Data, i) = d[i,:]
 # Convert geometries to Meshes.jl types
 # --------------------------------------
 
-coords2point(coords) = Point(coords)
-
-coords2multipoint(coords) = Multi(coords2point.(coords))
+function topoints(geom, is3d::Bool)
+  if is3d
+    [Point(GI.x(p), GI.y(p), GI.z(p)) for p in GI.getpoint(geom)]
+  else
+    [Point(GI.x(p), GI.y(p)) for p in GI.getpoint(geom)]
+  end
+end
 
-coords2chain(coords) = Chain(Point.(coords))
+tochain(geom, is3d::Bool) = Chain(topoints(geom, is3d))
 
-coords2multichain(coords) = Multi(coords2chain.(coords))
+function topolygon(geom, is3d::Bool)
+  exterior = topoints(GI.getexterior(geom), is3d)
+  if GI.nhole(geom) == 0
+      return PolyArea(exterior)
+  else
+    holes = map(g -> topoints(g, is3d), GI.gethole(geom))
+    return PolyArea(exterior, holes)
+  end
+end
 
-function coords2poly(coords)
-  chains = [Point.(coord) for coord in coords]
-  PolyArea(chains[begin], chains[begin+1:end])
+function GI.convert(::Type{Point}, ::GI.PointTrait, geom)
+  if GI.is3d(geom)
+    return Point(GI.x(geom), GI.y(geom), GI.z(geom))
+  else
+    return Point(GI.x(geom), GI.y(geom))
+  end
 end
 
-coords2multipoly(coords) = Multi(coords2poly.(coords))
-
-function geom2meshes(geom)
-  gtrait = GI.geomtrait(geom)
-  coords = GI.coordinates(geom)
-  if gtrait isa GI.PointTrait
-    coords2point(coords)
-  elseif gtrait isa GI.LineStringTrait
-    coords2chain(coords)
-  elseif gtrait isa GI.MultiLineStringTrait
-    coords2multichain(coords)
-  elseif gtrait isa GI.PolygonTrait
-    coords2poly(coords)
-  elseif gtrait isa GI.MultiPolygonTrait
-    coords2multipoly(coords)
+function GI.convert(::Type{Segment}, ::GI.LineTrait, geom)
+  p1, p2 = GI.startpoint(geom), GI.endpoint(geom)
+  if GI.is3d(geom)
+    Segment(Point(GI.x(p1), GI.y(p1), GI.z(p1)), Point(GI.x(p2), GI.y(p2), GI.z(p1)))
+  else
+    Segment(Point(GI.x(p1), GI.y(p1)), Point(GI.x(p2), GI.y(p2)))
```

Comment:
can you comment why we can't use Segment(topoints(geom)) here like with the other geometries? The Segment should accept a vector of points as well.

Reply:
Yes, it also works that way. Happy to modify that.